src/single-spa-alpinejs.js

```diff
@@ -1,20 +1,46 @@
-// Opts are passed into single-spa-alpinejs when creating an application/parcel
+/**
+ * Opts are passed into single-spa-alpinejs when creating an application/parcel
+ * template: (required) main template as string
+ * xData: (optional) externally provided x-data as an object or function. The single spa env props are passed to this if its a function
+ * xInit: (optional) externally provided x-init as a function it takes an argument as the dom element ID of the application/parcel root element. This will only be used if xData is also provided
+ */
 const defaultOpts = {
-  // required
-  template: null,
+  template: null, // required
+  xData: undefined,
+  xInit: undefined,
 };
 
 export default function singleSpaAlpineJs(opts) {
   opts = Object.assign({}, defaultOpts, opts);
 
   const typeofTemplate = typeof opts.template;
+  const typeofXData = typeof opts.xData;
+  const typeofXInit = typeof opts.xInit;
 
   if (typeofTemplate !== "function" && typeofTemplate !== "string") {
     throw Error(
       `single-spa-alpinejs: opts.template must be provided as a string or a function that returns a promise that resolves with a string`
     );
   }
 
+  // xData can be optional , but when provided needs to be either a function or an object
+  if (
+    typeofXData !== "undefined" &&
+    typeofXData !== "function" &&
+    typeofXData !== "object"
+  ) {
+    throw Error(
+      `single-spa-alpinejs: optional parameter opts.xData if provided must be as an object or a function that returns an object`
+    );
+  }
+
+  // xInit can be optional , but when provided needs to be a function
+  if (typeofXInit !== "undefined" && typeofXInit !== "function") {
```

Comment:
Same here regarding truthiness check. Also, we don't need to extract `typeofXInit` as a variable, but rather can just inline it in the if statement. The reason I extracted `typeofTemplate` is because of line 44 where it's used outside of the validation logic.

```suggestion
  if (opts.xInit && typeof opts.xInit !== "function") {
```